injector/cpuset_manager.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2022 Datadog, Inc.
+
+package injector
+
+import (
+	"fmt"
+	"github.com/DataDog/chaos-controller/cpuset"
+	"sync"
+)
+
+type StresserManager interface {
+	TrackInjectorCores(config CPUPressureInjectorConfig) error
+	GetCoresToBeStressed() cpuset.CPUSet
+	IsCoreAlreadyStressed(core int) bool
+	TrackStresser(core int, stresserPID int)
+	GetStresserPIDs() map[int]int
+}
+
+type cpuStressserManager struct {
+	mutex              sync.Mutex
+	coresToBeStressed  cpuset.CPUSet
+	stresserPIDPerCore map[int]int
+}
+
+func NewCPUStresserManager() StresserManager {
+	return &cpuStressserManager{
+		coresToBeStressed:  cpuset.NewCPUSet(),
+		stresserPIDPerCore: map[int]int{},
+	}
+}
+
+func (manager *cpuStressserManager) setCoresToBeStressed(cores cpuset.CPUSet) {
+	manager.mutex.Lock()
+	defer manager.mutex.Unlock()
+	manager.coresToBeStressed = cores
+}
+
+func (manager *cpuStressserManager) GetCoresToBeStressed() cpuset.CPUSet {
+	return manager.coresToBeStressed
+}
+
+func (manager *cpuStressserManager) IsCoreAlreadyStressed(core int) bool {
+	_, isCoreStressed := manager.stresserPIDPerCore[core]
```

Comment:
Maps are not thread safe in go, hence you need also to protect the read here.
Map being accessed by several goroutines, one could change the value when you are reading it.

Reply:
i'll add read lock as Aymeric has rightly suggested. Thanks both